src/app/hooks/useHealthCheck.ts

```diff
@@ -0,0 +1,48 @@
+import { useQuery } from "@tanstack/react-query";
+import { useEffect } from "react";
+
+import {
+  getHealthCheck,
+  isGeoBlocked,
+} from "@/app/services/healthCheckService";
+import { HealthCheckStatus } from "@/app/types/services/healthCheck";
+
+import { useError } from "../context/Error/ErrorContext";
+import { ErrorState } from "../types/errors";
+
+export const useHealthCheck = () => {
+  const { showError } = useError();
+
+  const { data, error, isError, refetch } = useQuery({
+    queryKey: ["api available"],
+    queryFn: getHealthCheck,
+    refetchOnMount: false,
+    refetchOnWindowFocus: false,
+  });
+
+  useEffect(() => {
+    if (isError) {
+      showError({
+        error: {
+          message: error.message,
+          errorState: ErrorState.SERVER_ERROR,
+          errorTime: new Date(),
+        },
+        retryAction: refetch,
+      });
+    }
+  }, [isError, error, showError, refetch]);
+
+  const isApiNormal = data?.status === HealthCheckStatus.Normal;
+  const isBlocked = data ? isGeoBlocked(data) : false;
+  const apiMessage = data?.message;
+
+  return {
+    isApiNormal,
+    isBlocked,
```

Comment:
```suggestion
    isGeoBlocked,
```

Reply:
`isGeoBlocked` is a funtion from `healthCheckService`, won't there be a confusion between this and hook return value? One is the boolean true/falce, another is a function that takes HealthCheckResult and returns a boolean

Reply:
Done via:

- `isGeoBlocked` inside the service -> `isGeoBlockedResult` (this function takes the `HealthCheckResult` hence the name)
- `isBlocked` -> `isGeoBlocked` - just a boolean value